Connectors/CSi/Speckle.Connectors.CSiShared/HostApp/CSiSendCollectionManager.cs

```diff
@@ -0,0 +1,42 @@
+using Speckle.Converters.Common;
+using Speckle.Converters.CSiShared;
+using Speckle.Sdk.Models.Collections;
+
+namespace Speckle.Connectors.CSiShared.HostApp;
+
+/// <summary>
+/// We can use the CSiWrappers to create our collection structure.
+/// </summary>
+/// <remarks>
+/// This class manages the collections. If the key (from the path) already exists, this collection is returned.
+/// If it doesn't exist, a new collection is created and added to the rootObject.
+/// </remarks>
+public class CSiSendCollectionManager
+{
+  private readonly IConverterSettingsStore<CSiConversionSettings> _converterSettings;
+  private readonly Dictionary<string, Collection> _collectionCache = new();
+
+  public CSiSendCollectionManager(IConverterSettingsStore<CSiConversionSettings> converterSettings)
+  {
+    _converterSettings = converterSettings;
+  }
+
+  // TODO: Frames could be further classified under Columns, Braces and Beams. Same for Shells which could be classified into walls, floors
+  public Collection GetAndCreateObjectHostCollection(ICSiWrapper csiObject, Collection rootObject)
```

Comment:
This could be renamed to `AddObjectCollectionToRoot`, which would be more descriptive.

Why is the root collection being returned if a new collection is created,
but the cached collection is returned when it is cached? Looks like the root collection is modified here, probably can just return void.

Reply:
Under commit "AddObjectCollectionToRoot":

- Agreed. Renamed to `AddObjectCollectionToRoot`
- True, unnecessary line removed.

Reply:
This comment probably applies to Revit too [Revit - SendCollectionManager.cs](https://github.com/specklesystems/speckle-sharp-connectors/blob/dev/Connectors/Revit/Speckle.Connectors.RevitShared/HostApp/SendCollectionManager.cs)